tests/mouse_test.go

```diff
@@ -5,30 +5,104 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-
-	"github.com/grafana/xk6-browser/common"
 )
 
-func TestMouseDblClick(t *testing.T) {
+func TestMouseActions(t *testing.T) {
 	t.Parallel()
 
-	b := newTestBrowser(t, withFileServer())
-	p := b.NewPage(nil)
-	opts := &common.FrameGotoOptions{
-		Timeout: common.DefaultTimeout,
-	}
-	_, err := p.Goto(
-		b.staticURL("dbl_click.html"),
-		opts,
-	)
-	require.NoError(t, err)
-
-	p.Mouse.DblClick(35, 17, nil)
-
-	v := p.Evaluate(`() => window.dblclick`)
-	bv := asBool(t, v)
-	assert.True(t, bv, "failed to double click the link")
-
-	got := p.InnerText("#counter", nil)
-	assert.Equal(t, "2", got)
+	t.Run("click", func(t *testing.T) {
+		t.Parallel()
+
+		tb := newTestBrowser(t)
+		p := tb.NewPage(nil)
+		m := p.GetMouse()
+
+		// Set up a page with a button that changes text when clicked
+		buttonHTML := `
+			<button onclick="this.innerHTML='Clicked!'">Click me</button>
+		`
+		p.SetContent(buttonHTML, nil)
+		button, err := p.Query("button")
+		require.NoError(t, err)
+
+		// Simulate a click at the button coordinates
+		box := button.BoundingBox()
+		m.Click(box.X, box.Y, nil)
+
+		// Verify the button's text changed
+		assert.Equal(t, "Clicked!", button.TextContent())
+	})
+
+	t.Run("double_click", func(t *testing.T) {
```

Comment:
I think this test should also ensure that a click has occurred twice. There are two things that need to occur:

1. `ondblclick` event is fired.
2. the element is clicked twice.

Before https://github.com/grafana/xk6-browser/pull/1121, the `ondblclick` event was fired, but the element wasn't being clicked on twice. That's why there was a counter in the test to ensure that the element was clicked on twice.

So we need to add something into the test to ensure that the element is in fact being clicked on twice too.

Reply:
👍 [8beba15](https://github.com/grafana/xk6-browser/pull/1318/commits/8beba159aaf0f124f72c45404bd3035a3d38cc8e)